### PR TITLE
adding possibility to specify custom repositories for coursier (#1521)

### PR DESCRIPTION
**Pull request by slivkamiro**
I'm kind of lost in tests. I would need help with writing a test for ScalafmtDynamicDownloader, if necessary (I haven't found any).
I'm also struggling with sbt-scalafmt part. I can't find a way how to map resolvers configuration into list of string urls. Any ideas?

### CODE REVIEW

File: scalafmt-dynamic/src/main/scala/org/scalafmt/dynamic/ScalafmtDynamicDownloader.scala [hunk 1]
```diff
@@ -91,7 +91,7 @@ class ScalafmtDynamicDownloader(
         "https://oss.sonatype.org/content/repositories/snapshots"
       ),
       MavenRepository.of("https://oss.sonatype.org/content/repositories/public")
-    )
+    ) ++ customRepositories
```
Comment (tomasherman) [inline scalafmt-dynamic/src/main/scala/org/scalafmt/dynamic/ScalafmtDynamicDownloader.scala]: i think custom repos should either override the defaults, or have higher priority - one usecase of this (ours) is that oss repositories are disabled and must go through our own artifactory and it causes timeouts when trying to download stuff from OSS ... therefore it makes sense to try custom repositories first and fallback on defaults

Comment (slivkamiro) [reply scalafmt-dynamic/src/main/scala/org/scalafmt/dynamic/ScalafmtDynamicDownloader.scala]: timeouts are good point, will move it in front then

**Comment by sideeffffect** [comment]
@slivkamiro thank you so much for this!
I know this is just `scalafmt` proper, but are you perhaps considering implementing support for this in `sbt-scalafmt` plugin? It would be great, if the `-Dsbt.repository.config=/.../repositories -Dsbt.override.build.repos=true` overrides were respected.

File: scalafmt-interfaces/src/main/java/org/scalafmt/interfaces/Scalafmt.java [hunk 2]
```diff
@@ -74,6 +73,11 @@
      */
     Scalafmt withReporter(ScalafmtReporter reporter);
 
+    /**
+     * Use this repositories to resolve dependencies.
+     */
+    Scalafmt withRepositories(String ... repositories);
```
Comment (olafurpg) [inline scalafmt-interfaces/src/main/java/org/scalafmt/interfaces/Scalafmt.java]: Let's make it clear that these must be Maven repositories
```suggestion
    Scalafmt withMavenRepositories(String ... repositories);
```

Comment (slivkamiro) [reply scalafmt-interfaces/src/main/java/org/scalafmt/interfaces/Scalafmt.java]: no problem, I will amend the commit if you don't mind

**Comment by slivkamiro** [comment]
> @slivkamiro thank you so much for this!
> I know this is just `scalafmt` proper, but are you perhaps considering implementing support for this in `sbt-scalafmt` plugin? It would be great, if the `-Dsbt.repository.config=/.../repositories -Dsbt.override.build.repos=true` overrides were respected.

Yes, see https://github.com/scalameta/sbt-scalafmt/pull/73

**Pull request merged**